app/src/AddItemForm.js

```diff
@@ -0,0 +1,103 @@
+import "./App.css";
+
+import React, { useState, useEffect, useRef } from "react";
+
+import * as apiClient from "./apiClient";
+
+const AddItem = ({ onAdd }) => {
+  const inputItem = useRef();
+  const [name, setName] = useState("");
+  console.log(name);
+
+  const addNewItem = async (name) => {
+    const response = await apiClient.addItem(name);
+    onAdd(response[0]);
+    // window.location = "/";
+    console.log(response);
+  };
+  // Clicking Edit/on the value activates editMode. User sees input bar and Submit button.
+  const onEditClick = () => {
+    setEditMode(true);
+  };
+
+  // Submit sets setEditMode(false). User sees value with submitted input item.
+  const onSubmitClick = () => {
+    addNewItem(name);
+    setName("");
+    setEditMode(false);
+  };
+
+  // Want: When user clicks out => setEditMode(false) => editMode === false, a line with original value.
+  //   const cancelEdit = () => {
+  //   setEditMode(false);
+  // }
+
+  // When user clicks Enter on Edit Mode, onSubmitClick is processed.
+  const handleKeyPress = (e) => {
+    if (e.charCode === 13) {
+      addNewItem(e.target.value);
+      setName("");
+      setEditMode(false);
+    }
+  };
+
+  // By default, editMode starts out as false.
+  const [editMode, setEditMode] = useState(false);
+  useEffect(() => {
+    if (editMode) {
+      inputItem.current.focus();
+    }
+  }, [editMode]);
+
+  return (
+    <>
+      {!editMode && (
+        <div
+          className="add-item-row"
+          onClick={() => {
+            onEditClick();
+          }}
+          onKeyPress={() => {
+            onEditClick();
```

Comment:
I noticed that if you remove this onKeyPress method, the "press Enter to save" functionality works. I think this keypress was being processed to setEdit to be true, when you wanted it to be false. Can you confirm that you still need this functionality?

Reply:
When I remove it, my linter tells me that if the div has an onClick() method, it also requires a key event for those who might need to tab through the page to interact with it in the case they cannot use a mouse! When I remove the line, the code won't compile for me.

Reply:
Makes sense! It's working for me now, which is great.